### PR TITLE
Signalr support

### PR DESCRIPTION
**Pull request by olalid**
Add support for signalr stream data

### CODE REVIEW

File: pyeasee/easee.py [hunk 1]
```diff
@@ -182,6 +198,146 @@ async def close(self):
...
+            .build()
+        )
+        self.sr_connection.on_open(lambda: self._sr_open_cb())
+        self.sr_connection.on_close(lambda: self._sr_close_cb())
+        self.sr_connection.on("ProductUpdate", self._sr_product_update_cb)
+        self.sr_connection.on("ChargerUpdate", self._sr_charger_update_cb)
+
+        await self._verify_updated_token()
+        while True:
+            """ The siglarcore lib start function does blocking I/O, so can not be called directly """
```
Comment (Hellowlol) [inline pyeasee/easee.py:297]: signalcore

File: pyeasee/utils.py [hunk 2]
```diff
@@ -1,11 +1,42 @@
...
+from .const import ChargerStreamData, EqualizerStreamData
 
 regex = r"^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\.[0-9]+)?(Z|[+-](?:2[0-3]|[01][0-9]):[0-5][0-9])?$"
 match_iso8601 = re.compile(regex).match
 
 
+def lookup_charger_stream_id(id):
+    try:
+        name = ChargerStreamData(id).name
+    except BaseException:
```
Comment (Hellowlol) [inline pyeasee/utils.py:13]: IMO this is too broad exception. Consider using ValueError

File: pyeasee/utils.py [hunk 3]
```diff
@@ -1,11 +1,42 @@
...
+        return None
+    return name
+
+
+def lookup_equalizer_stream_id(id):
+    try:
+        name = EqualizerStreamData(id).name
+    except BaseException:
```
Comment (Hellowlol) [inline pyeasee/utils.py:21]: IMO this is too broad exception. Consider using ValueError

File: pyeasee/easee.py [hunk 4]
```diff
@@ -182,6 +198,146 @@ async def close(self):
...
 
+        await self._sr_disconnect()
+
+    def _sr_token(self):
+        """
+        Return access token to signalr library, called from signalr thread, internal use only
+        """
+        if self.running_loop is not None:
+            future = asyncio.run_coroutine_threadsafe(self._verify_updated_token(), self.event_loop)
+            future.result()
```
Comment (Hellowlol) [inline pyeasee/easee.py:209]: Should we wait for the future to be done?

Comment (olalid) [reply pyeasee/easee.py:209]: future.result() does actually wait until the function is done..
(This is called from the the signalrcore synchronous lib thread so we can not use await).
See https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future

File: pyeasee/utils.py (hunk 2 above)
Comment (olalid) [reply pyeasee/utils.py:13]: Yes, will fix.

File: pyeasee/utils.py (hunk 3 above)
Comment (olalid) [reply pyeasee/utils.py:21]: Yes, will fix.

File: pyeasee/easee.py (hunk 1 above)
Comment (olalid) [reply pyeasee/easee.py]: Fixed

**Pull request merged**